### ISSUE TITLE
Create Docker development environment

### ISSUE THREAD

**Issue by mrbiggred** [corgibytes/freshli-lib#304]
Some work was done to create a Docker development environment for #238 but I think this only works for VS Code.  We should have a proper Docker development environment that uses Docker and Docker Compose.  Ideally anyone should be able to build the development environment with:

```
docker-compose build
```

This built environment should work on Windows, Mac, and Linux.  It should also work with most IDEs including Visual Studio 2019, VS Code, and Rider.

The Docker environment should be built using Docker [BuildKit](https://docs.docker.com/develop/develop-images/build_enhancements/).  I believe Docker BuildKit is enabled by default on Windows and Mac.  On Linux you need to set the following environment variables:

```
# .profile
export DOCKER_BUILDKIT=1
export COMPOSE_DOCKER_CLI_BUILD=1
```

The other thing you need to do is set the `#syntax` at the top of the Dockerfile:

```dockerfile
# syntanx=docker/dockerfile:1.2
```

More info [here](https://docs.docker.com/engine/reference/builder/#syntax) and [here](https://hub.docker.com/r/docker/dockerfile/).

**Comment by marnen** [comment]
@RepoCorp Have you done any work on this? If not, I'm willing to try to get this to happen.

What's a good way to verify that the dev setup works? Run `dotnet test`?

**Comment by RepoCorp** [comment]
@marnen  I didn't do much back then and it's been a while. Go for it.

**Comment by mscottford** [comment]
If you look in the https://github.com/corgibytes/freshli-lib/tree/main/.devcontainer directory, you can see the container specification that I've been using for working with GitHub Codespaces ( https://github.com/features/codespaces ).

M. Scott Ford
Co-Founder & Chief Code Whisperer (CTO)
Corgibytes, LLC
804.596.2375 x701
pronouns: he/him
***@***.***
https://corgibytes.com ( https://corgibytes.com/ )

Have you read our First Round Review ( http://firstround.com/review/forget-technical-debt-heres-how-to-build-technical-wealth/ ) article about paying off technical debt?

Love refactoring and TDD? Join us at LegacyCode.Rocks ( http://LegacyCode.Rocks ) for virtual meetups, podcasts, and more.

Sent via Superhuman ( ***@***.*** )

**Comment by marnen** [comment]
@mscottford Yeah, I looked in there. It looked like the Docker configuration was specific enough to VS Code that it wouldn't be useful for a general Docker environment, but perhaps I misunderstood something.

I didn't know about Codespaces, which seems kind of cool (but still VS Code, so not _all_ that helpful in decoupling from it).

If you have any suggestions about the work I did in https://github.com/marnen/freshli-lib/commit/21141adbb9b45df36b4f02f4392d5489f240869d, I'd be really curious. I don't really know enough about .NET to know what I'm doing beyond the most cursory level here. :)
